content/posts/2022-09-14-introducing-jaiminho/index.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Introducing Jaiminho, an implementation of the Transactional Outbox Pattern for Django
+author: Rafael Zanetti
+twitter: LoadsmartEng
+layout: post
+lang: en
+path: /blog/introducing-jaiminho
+date: 2022-11-25
+comments: true
+---
+
+At Loadsmart, many of our services are organized within an event-driven architecture and face some usual challenges that come with such an architecture. One of them is **dual writes**, which happens when an application has to change data on two different places. A common example of dual writes is an application that needs to store data on a database and notify that change by emitting an event which may be consumed by different services. The inherent issue with this scenario is that if one of these two operations fails, the producer and consumer applications will have data inconsistencies. If we choose to write the change to the database first and then notify the event, in the case of a notification failure the database would have the updated data and the consumer wouldn’t. If we reverse it (i.e. notifying first and then storing), a failure when committing the transaction in the database would cause the consumer receiving data that isn’t on the application database.
+
+There are well-known methods to deal with such issues, and we chose to apply the transactional outbox pattern. Since we couldn’t find any related solution on [Django community](https://djangopackages.org/search/?q=outbox), we decided to implement an open-source library called [Jaiminho](https://github.com/loadsmart/jaiminho), which is a broker agnostic implementation of the outbox pattern.
+
+
+# The Transactional outbox pattern
```

Comment:
**suggestion**: What about adding an image to describe how it works?

Reply:
Nice idea! Will draw it and include